pages/model/[uuid].tsx

```diff
@@ -108,12 +111,47 @@ function Model() {
     setCopyModelCardSnackbarOpen(false)
   }
 
+  useEffect(() => {
+    if (currentUser) {
+      if (currentUser !== undefined && currentUser.id === version?.metadata.contacts.manager) {
+        updateLastViewed('manager')
+      }
+      if (currentUser !== undefined && currentUser.id === version?.metadata.contacts.reviewer) {
+        updateLastViewed('reviewer')
+      }
+    }
+  }, [managerLastViewed, reviewerLastViewed, currentUser])
+
+  const updateLastViewed = (role: string) => {
+    if (currentUser?.id === version?.metadata.contacts.manager) {
+      const managerLastViewedDate = new Date(managerLastViewed as Date)
+      const reviewerLastViewedDate = new Date(reviewerLastViewed as Date)
+      const updatedAtDate = new Date(version!.updatedAt)
+      if (
+        managerLastViewedDate.getTime() < updatedAtDate.getTime() ||
+        reviewerLastViewedDate.getTime() < updatedAtDate.getTime()
+      ) {
+        setShowLastViewedWarning(true)
+      }
+      putEndpoint(`/api/v1/version/${version!._id}/lastViewed/${role}`)
+    }
+  }
+
   useEffect(() => {
     if (tab !== undefined && isTabOption(tab)) {
       setGroup(tab)
     }
   }, [tab])
 
+  useEffect(() => {
```

Comment:
Using non-null assertion and missing dependencies, can also be simplified slightly:
```
  useEffect(() => {
    if (version && !managerLastViewed) {
      setManagerLastViewed(version.managerLastViewed)
    }
    if (version && !reviewerLastViewed) {
      setReviewerLastViewed(version.reviewerLastViewed)
    }
  }, [version, managerLastViewed, reviewerLastViewed])
```